compiler/quilt/tools/core.py

```diff
@@ -59,20 +59,15 @@ def __json__(self):
     def get_children(self):
         return {}
 
-    def preorder(self, sort=False):
+    def preorder(self):
         """
         Iterator that returns all nodes in the tree starting with the current node.
-
-        :param sort: within each group, sort child nodes by name
         """
         stack = [self]
         while stack:
             obj = stack.pop()
             yield obj
-            if sort:
-                stack.extend(child for name, child in sorted(iteritems(obj.get_children()), reverse=True))
-            else:
-                stack.extend(itervalues(obj.get_children()))
+            stack.extend(itervalues(obj.get_children()))
```

Comment:
so what guarantees order now that there's no call to `sorted`?

Reply:
i mean `preorder` implies RLV and user might get LRV etc with this method?

Reply:
it's fine if we want to loosen the contract but we should be explicit about it.

Reply:
There's only one place we use this method now, and it doesn't care about the order. In fact, we really don't need a preorder method.